### PR TITLE
CI/ fix binary building docker

### PR DESCRIPTION
**Pull request by rok-cesnovar**
Static binary docker also needs to pin the js_of_ocaml version. Trying it out first.

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
Any help or hint with this would be appreciated. Thanks!

**Comment by seantalts** [comment]
This is making me realize we should probably have the PRs build the static binaries even though they won't upload them anywhere. I guess they could also upload them to Jenkins or something.

**Comment by rok-cesnovar** [comment]
Thanks @seantalts !

This seems to work fine now: https://jenkins.mc-stan.org/blue/organizations/jenkins/stanc3/detail/PR-701/9/pipeline/198/

I uncommented it back now.

**Comment by rok-cesnovar** [comment]
> This is making me realize we should probably have the PRs build the static binaries even though they won't upload them anywhere. I guess they could also upload them to Jenkins or something.

I agree. Lets just not publish them for now until we think of how/where to put them.

For builiding and publishin binaries it think the stanc-3test-binaries job is fine: https://jenkins.mc-stan.org/job/stanc3-test-binaries/build?delay=0sec

File: scripts/install_dev_deps.sh [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 # Merlin, utop, ocp-indent, ocamlformat, and patdiff are all for developer assistance
 opam pin -y ocamlformat 0.8
-opam install -y merlin utop ocp-indent patdiff
+opam install -y ocamlformat.0.8 merlin utop ocp-indent patdiff
```
Comment (seantalts) [inline scripts/install_dev_deps.sh:5]: I think `pin`ing ocamlformat should also install it, but I'm going to merge this for the fix anyway :)

**Review by seantalts — APPROVED** [review]

**Comment by seantalts** [comment]
Thank you!

**Pull request merged**